docs/SdkVersioning.md

```diff
@@ -0,0 +1,590 @@
+```
+
+```mermaid
+gitGraph
+  commit id: "0.1.0-dev1"
+  commit id: "0.1.0-dev2"
+  branch feat/build-feature
+  checkout feat/build-feature
+  commit id: "commit-1"
+  checkout main
+  commit id: "0.1.0-dev3"
+  checkout feat/build-feature
+  commit id: "commit-2"
+  checkout main
+  merge feat/build-feature id: "0.1.0-dev4" type: HIGHLIGHT
+```
+
+Once the Pull Request is merged, a new `draft` GitHub release (`v0.1.0-dev4`) will be created;
+
+```sh
+$ gh release list
+
+TITLE           TYPE    TAG NAME      PUBLISHED
+v0.1.0-dev4     Draft   v0.1.0-dev4   about 1 minute ago
+```
+
+## Release Candidate flow
+
+### Workflow
+Create a workflow for managing our development releases:
+
+`.github/workflows/release-candidate.yml`:
+```yml
+name: Promote to Release Candidate
+on:
+  workflow_dispatch:
+
+jobs:
+  commisery-action:
+    name: Promote to Release Candidate
+    runs-on: ubuntu-latest
+    steps:
+      - name: Create Release Candidate
+        uses: tomtom-international/commisery-action@v2
+        with:
+          token: ${{ github.token }}
+          create-release: true     # Create a GitHub Release
+          version-prefix: v        # Adds the prefix `v` to the version (i.e. `v.0.1.0`)
+          sdkver-release-type: rc  # Promote latest Development release to Release Candidate
+```
+
+### Promoting a Development Release to Release Candidate
+
+Considering the current state:
+
+```mermaid
+gitGraph
+  commit id: "0.1.0-dev1"
+  commit id: "0.1.0-dev2"
+
+  branch feat/build-feature
+  checkout feat/build-feature
+  commit id: "commit-1"
+  checkout main
+  commit id: "0.1.0-dev3"
+  checkout feat/build-feature
+  commit id: "commit-2"
+  checkout main
+  merge feat/build-feature id: "0.1.0-dev4"
+```
+
+One can trigger a promotion of the latest Development Release to a Release Candidate by 
+running the created worklow:
+
+```sh
+$ gh workflow run release-candidate.yml
+```
+
+This will create a new *tagged* GitHub Release with the `pre-release` flag enabled:
+
+```mermaid
+gitGraph
+  commit id: "0.1.0-dev1"
+  commit id: "0.1.0-dev2"
+  branch feat/build-feature
+  checkout feat/build-feature
+  commit id: "commit-1"
+  checkout main
+  commit id: "0.1.0-dev3"
+  checkout feat/build-feature
+  commit id: "commit-2"
+  checkout main
+  merge feat/build-feature id: "0.1.0-dev4" tag: "v0.1.0-rc1" type: HIGHLIGHT
+```
+
+```sh
+$ gh release list
+
+TITLE           TYPE          TAG NAME      PUBLISHED
+v0.1.0-rc1      Pre-release   v0.1.0-rc1    about 2 minutes ago
+```
+
+### Hardening a Release Candidate
+
+Considering the following initial state:
+```mermaid
+gitGraph
+  commit id: "0.1.0-dev1"
+  commit id: "0.1.0-dev2"
+  commit id: "0.1.0-dev3"
+  commit id: "0.1.0-dev4" tag: "v0.1.0-rc1"
+  commit id: "0.2.0-dev1"
+  commit id: "0.2.0-dev2"
+  commit id: "0.2.0-dev3"
+  commit id: "0.2.0-dev4"
+```
+
+In order to harden a Release Candidate, one needs to:
+- Create a `release`-branch based on the specified release candidate
```

Comment:
Reading through the flow I would suggest we make this part of the suggested workflow already. As the workflow is already quite specific in its action right now to promote from devN to rc1, as otherwise one could have decided better to mark devN as release directly.

Reply:
Although I agree that automation would be beneficial (and will prevent potential issues), I question that commisery-action is the rightful owner of this.